src/transaction/builder/tx_data.rs

```diff
@@ -0,0 +1,115 @@
+use std::collections::HashSet;
+
+use ckb_types::{
+    core::{self, TransactionBuilder},
+    packed::{self, CellDep, CellInput, CellOutput, Script},
+    prelude::*,
+};
+
+use crate::tx_builder::TxBuilderError;
+
+#[derive(Default)]
+pub struct TxData {
```

Comment:
please ckb_types::core::TransactionBuilder : https://github.com/nervosnetwork/ckb/blob/v0.109.0/util/types/src/core/advanced_builders.rs#L20, this TxData struct is unnessary.

Reply:
The cell_deps may add several times, TransactionBuilder doesn't have the ability to do deduplication, so here i use a new data structure.

Reply:
We should create a PR in ckb repo to improve the usability instead of repeat implementing similar struct / fn here if the dedup is the only issue.

Reply:
For now, i think we should keep the TxData struct, at least it can get correct result, after a patch fix this problem, then it can be removed.

Reply:
Now i create a core::TransactionBuilder in the SDK, and add necessary function in the class.